fdbserver/DataDistribution.actor.cpp

```diff
@@ -2569,6 +2585,188 @@ ACTOR Future<Void> waitUntilHealthy(DDTeamCollection* self, double extraDelay =
 	}
 }
 
+// Take a snapshot of necessary data structures from `DDTeamCollection` and print them out with yields to avoid slow
+// task on the run loop. This function is very expensive to call due to the data it needs to copy. Use with caution.
+ACTOR Future<Void> printSnapshotTeamsInfo(Reference<DDTeamCollection> self, std::string tcIdString) {
+	state DatabaseConfiguration configuration;
+	state std::map<UID, Reference<TCServerInfo>> server_info;
+	state std::map<UID, ServerStatus> server_status;
+	state vector<Reference<TCTeamInfo>> teams;
+	state std::map<Standalone<StringRef>, Reference<TCMachineInfo>> machine_info;
+	state std::vector<Reference<TCMachineTeamInfo>> machineTeams;
+	// state std::vector<std::string> internedLocalityRecordKeyNameStrings;
+	// state int machineLocalityMapEntryArraySize;
+	// state std::vector<Reference<LocalityRecord>> machineLocalityMapRecordArray;
+	state int traceEventsPrinted = 0;
+	state std::vector<const UID*> serverIDs;
+	loop {
+		wait(self->printDetailedTeamsInfo.onTrigger());
+
+		traceEventsPrinted = 0;
+		if (now() - self->lastTeamCollectionInfoSnapshotTime >= SERVER_KNOBS->DD_TEAMS_INFO_SNAPSHOT_REFRESH_INTERVAL) {
+			// Refresh state snapshot
+			self->lastTeamCollectionInfoSnapshotTime = now();
+
+			configuration = self->configuration;
+			server_info = self->server_info;
+			teams = self->teams;
+			machine_info = self->machine_info;
+			machineTeams = self->machineTeams;
+			// internedLocalityRecordKeyNameStrings = self->machineLocalityMap._keymap->_lookuparray;
+			// machineLocalityMapEntryArraySize = self->machineLocalityMap.size();
+			// machineLocalityMapRecordArray = self->machineLocalityMap.getRecordArray();
+			std::vector<const UID*> _uids = self->machineLocalityMap.getObjects();
+			serverIDs = _uids;
+
+			auto const& keys = self->server_status.getKeys();
+			for (auto const& key : keys) {
+				server_status.emplace(key, self->server_status.get(key));
+			}
+		}
+
+		// Print to TraceEvents
+		TraceEvent("DDConfig")
```

Comment:
Change `TraceEvent("DDConfig")` event to `TraceEvent("DDConfig", self->distributorId).detail("Primary", self->primary)`

Please add the `distributorId ` and `self->primary` to the other trace events in this actor.

Reply:
Sure. Every trace event already has the `TeamCollectionID` set, but that's a string field. I will add `self->primary` just to be safe.